### PR TITLE
arch: nios2: enable System ID soft IP driver

### PR DESCRIPTION
**Pull request by ramakrishnapallala**
The system ID core is a simple read-only device that provides Qsys systems with a unique identifer.

Nios-II processor systems use the system ID core to verify that an executable program was compiled targeting the actual hardware image configured in the target FPGA.

### CODE REVIEW

File: samples/boards/altera_max10/sysid/src/main.c [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#include <zephyr.h>
+#include <misc/printk.h>
+#include "altera_avalon_sysid.h"
+
+void main(void)
+{
+	s32_t sysid;
+
+	sysid = alt_avalon_sysid_test();
+	if (!sysid)
```
Comment (andrewboie) [inline samples/boards/altera_max10/sysid/src/main.c]: zephyr coding style requires the use of braces at all times

File: samples/boards/altera_max10/sysid/README.rst [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+   :zephyr-app: samples/boards/altera_max10/sysid
+   :board: altera_max10
+   :goals: build
+   :compact:
+
+Sample Output
+=============
+
+nios2 core output
+---------------
```
Comment (dbkinder) [inline samples/boards/altera_max10/sysid/README.rst]: underline needs to be at least as long as the title (add some more ----s)

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: samples/boards/altera_max10/sysid/src/main.c (hunk 1 above)
Comment (ramakrishnapallala) [reply samples/boards/altera_max10/sysid/src/main.c]: oops!! sorry i missed it.

File: samples/boards/altera_max10/sysid/README.rst (hunk 2 above)
Comment (ramakrishnapallala) [reply samples/boards/altera_max10/sysid/README.rst]: Ok.

File: samples/boards/altera_max10/sysid/src/main.c [hunk 3]
```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <misc/printk.h>
+#include "altera_avalon_sysid.h"
```
Comment (dbkinder) [inline samples/boards/altera_max10/sysid/src/main.c]: no such file?

**Review by dbkinder — CHANGES_REQUESTED** [review]
+1 for the docs, but there's a missing include file altera_avalon_sysid.h

File: samples/boards/altera_max10/sysid/src/main.c (hunk 3 above)
Comment (ramakrishnapallala) [reply samples/boards/altera_max10/sysid/src/main.c]: this sample is only applicable to altera_max10 board. somehow it is being built for qemu_x86 by shippable. I only added altera_max10 under platform_whitelist in sample.yaml so it should not build for other targets.

Comment (ramakrishnapallala) [reply samples/boards/altera_max10/sysid/src/main.c]: @dbkinder now it's fine, shippable is passed.

**Review by dbkinder — APPROVED** [review]

File: samples/boards/altera_max10/sysid/src/main.c [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+void main(void)
+{
+	s32_t sysid;
+
+	sysid = alt_avalon_sysid_test();
+	if (!sysid) {
+		printk("[SysID] hardware and software appear to be in sync\n");
```
Comment (andrewboie) [inline samples/boards/altera_max10/sysid/src/main.c]: If this is the expected output, can we just turn this into a test case and not a sample, where success/failure is reported via the typical test case way?
We could then remove the build_only tag.

Comment (ramakrishnapallala) [reply samples/boards/altera_max10/sysid/src/main.c]: do you want me to move this to ztest framework?

Comment (andrewboie) [reply samples/boards/altera_max10/sysid/src/main.c]: what I want is for this to be tested at runtime, and report pass/fail like other test cases, and get rid of build_only
This should be a test, not a sample

Comment (nashif) [reply samples/boards/altera_max10/sysid/src/main.c]: put it in tests/boards/....

**Comment by ramakrishnapallala** [comment]
@andrewboie , updated the PR please look into it.

**Review by tyrantlizard77 — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**